### PR TITLE
refactor use of container struct from daemon

### PR DESCRIPTION
**Pull request by MHBauer**
- use Exists instead of Get
- push Get inside of daemonbuilder

like #16366 

@calavera @duglin @icecrime

### CODE REVIEW

File: api/server/router/local/image.go [hunk 1]
```diff
@@ -63,12 +63,11 @@ func (s *router) postCommit(ctx context.Context, w http.ResponseWriter, r *http.
 		Config:  c,
 	}
 
-	container, err := s.daemon.Get(cname)
-	if err != nil {
+	if !s.daemon.Exists(cname) {
 		return err
```
Comment (thaJeztah) [inline api/server/router/local/image.go]: possibly n00b comment; but is this err related to this "if"?

Comment (MHBauer) [inline api/server/router/local/image.go]: no, that's wrong. I need to take another look. Should probably be a derr.containernotexists or somesuch.

**Comment by MHBauer** [comment]
thanks @thaJeztah good catch.

**Comment by calavera** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐻
/ping @tiborvass

**Comment by duglin** [comment]
LGTM
